RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,301 @@
+
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+following conditions are met:
+
+1. Redistributions of this document must retain the above copyright notice, this list of conditions and the following
+   disclaimer.
+2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+   disclaimer in the documentation and/or other materials provided with the distribution.
+3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products
+   derived from this software without specific prior written permission.
+
+THIS DOCUMENT IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", 
+"NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in 
+[BCP 14](https://tools.ietf.org/html/bcp14) (covering RFC2119 and RFC8174) when, and only when, they appear in all capitals, as 
+shown here.
+
+## Disclaimer
+
+The purpose of this document and its content is for information purposes only and may be subject to change or update
+without notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community regarding the
+technological merits of the potential system outlined herein.
+
+## Goals
+
+This document will introduce the Tari communication network and the communication protocol used to select, establish and maintain connections between peers on the network.
+[Consensus Node]s and [Consensus Client]s will be introduced and their required functionality will be proposed.
+
+## Related RFCs
+
+* [RFC-0100: The Tari Base Layer](RFC-0100_BaseLayer.md)
+* [RFC-0300: Digital asset network](RFC-0300_DAN.md)
+
+## Description
+
+### Assumptions
+
+- A communication channel can be established between two peers once their online communication addresses are known to each other.
+- A [Validator Node] is able to obtain a [node id] from the registration process on the Base Layer.
```

Comment:
If in agreement, please capitalize ID everywhere in the text.

Reply:
Applied change everywhere.

Reply:
I'm not sure the node ID is obtained from the "registration information" because it is not recorded on the Base Layer. The node ID is calculated from the location where the Validator Nodes registration appears on the Base Layer. That is why I believed that "registration process" was a better description.

Reply:
To obtain something from a "_process_" active participation is required. 
Maybe state "_A [Validator Node] is able to derive a [node ID] from the registered location on the Base Layer._"?

Reply:
Change applied.